src/App.ts

```diff
@@ -568,18 +574,22 @@ function singleTeamAuthorization(
   authorization: Partial<AuthorizeResult> & { botToken: Required<AuthorizeResult>['botToken'] },
 ): Authorize {
   // TODO: warn when something needed isn't found
-  const botUserId: Promise<string> = authorization.botUserId !== undefined ?
-    Promise.resolve(authorization.botUserId) :
+  const auth: Promise<AuthConstraints> = authorization.botUserId !== undefined && authorization.botId !== undefined ?
+    Promise.resolve({ botUserId: authorization.botUserId, botId: authorization.botId }) :
     client.auth.test({ token: authorization.botToken })
-      .then(result => result.user_id as string);
-  const botId: Promise<string> = authorization.botId !== undefined ?
-    Promise.resolve(authorization.botId) :
-    botUserId.then(id => client.users.info({ token: authorization.botToken, user: id }))
-      .then(result => ((result.user as any).profile.bot_id as string));
-  return async () => ({
-    botToken: authorization.botToken,
-    botId: await botId,
-    botUserId: await botUserId,
+      .then((result) => {
+        return {
+          botUserId: (result.user_id as string),
+          botId: (result.bot_id as string),
+        };
+      });
+
+  return () => auth.then((result) => {
```

Comment:
small nitpick. i think the following would be eaiser to read (because I think `.then()` should be used sparingly when there's something `await` doesn't suit).

```
return async () => {
  return Object.assign({ botToken: authorization.botToken }, await auth);
};
```